### DISCUSSION TITLE
Unable to upgrade to 8.5 and up

### DISCUSSION THREAD

**Discussion by RolandTalstra** [arendst/Tasmota#19634] [Support request] [answered]
Hi
I've been trying to upgrade from current version 8.1.0 to 8.5.1 (and from there to latest, for use in Home Assistant)

Unfortunately when trying to upgrade from URL I get  "Upgrade":"Failed HTTP error: connection refused"
and when I try to do it from local file, I get : Upload Failed  Magic byte is not 0xE9

I'm a bit at a loss what to do here. Any suggestions?

**Comment by barbudor** [comment]
What exact URL have you entered?
Are you sure that your device has access to the internet? (proper gateway and dns settings)

**Reply by RolandTalstra** [reply]
Thanks for your reply.
The URL i've used: http://ota.tasmota.com/tasmota/release-8.5.1/tasmota.bin)
My device is able to connect to outside my network.

**Comment by Jason2866** [comment, marked as answer]
Local downloaded file is probably corrupted. For esp8266 Devices use only the zipped files (extension "gz")
For esp32 use only the files **WITHOUT** factory in the name.

**Reply by sfromis** [reply]
Beware that upgrading **from** version 8.1 will not support `.gz` files yet, that feature started next version.

**Reply by RolandTalstra** [reply]
I've tried several files, and several downloads of those files. Just the .bin cause i'm on the old version.

Haven't seen any file with factory in the name tbh

**Reply by sfromis** [reply]
You may not have downloaded files the correct way. Yes, that's plausible. Do not do anything like right-click and "save as", only click on the link.

And `.factory` files are only for ESP32 devices, which is probably not what you have.

**Reply by RolandTalstra** [reply]
ok, yeah i clicked on the file, and the browser downloaded it subsequently

**Reply by sfromis** [reply]
Well, the explanation of the magic byte error is still that Tasmota did a simple validation of the file, and found it to be incorrect. Better verify the file, at least check for correct size. What Tasmota validates is that the first byte of the file is 0xE9 for `.bin`, or 0x1F for `.bin.gz`.

**Reply by RolandTalstra** [reply]
Ah, thanks for that explanation. Ill keep trying several sources.

**Reply by sfromis** [reply]
It may not be about trying other sources, but about how the files are handled

**Reply by RolandTalstra** [reply]
well, I've downloaded the file through several browsers. The filesize corresponds, so I'm not sure what else I can do in terms of verifying the file

**Reply by sfromis** [reply]
Well, look at the content, like using some hex editor, and check first byte.

**Reply by barbudor** [reply]
What I don't get is why you are not trying to follow the [upgrade path](https://tasmota.github.io/docs/Upgrading/#upgrade-flow) documentation ?

It is not required to download any file but use the URL from the upgrade path directly in the "Upgrade by Web Server / OTA URL" entry
The first one not being the URL you have listed but http://ota.tasmota.com/tasmota/release-8.5.1/tasmota-lite.bin

**Reply by RolandTalstra** [reply]
I did follow that documentation.  I've mentioned 1 URL above, but I've tried several. 
Not sure how and why it didn't work,  probably made some errors there.

Your exact URL worked, so, now I've upgraded to 13.1.0 (in steps)

So, thanks for your help !

**Reply by barbudor** [reply]
It is important to use the exact URL which provided in that documentation
The 'lite' variant can be flashed in single step (avoiding the intermediate minimal step)
You can then upgrade to the full Tasmota at latest step
